### PR TITLE
use include_eds in config_dump for proxy

### PR DESCRIPTION
**Pull request by shamsher31**
Envoy API https://github.com/envoyproxy/envoy/pull/11425 also provided EDS output for `config_dump` including [endpoint_config](https://www.envoyproxy.io/docs/envoy/latest/api-v3/admin/v3/config_dump.proto#admin-v3-endpointsconfigdump).

```json
{
  "static_endpoint_configs": [],
  "dynamic_endpoint_configs": []
}
```

### CODE REVIEW

**Comment by shamsher31** [comment]
@howardjohn I have updated `go-control-plane v0.9.6` still not able to see the response `endpoint_config` in `config_dump`. Is there any other dependency needs to be updated ?

**Comment by lizan** [comment]
the image from istio/proxy master should be already new enough contains the dump

**Comment by shamsher31** [comment]
@howardjohn  I noticed `istio/proxy` is still using [go-control-plane v0.9.5](https://github.com/istio/proxy/blob/master/go.mod#L11) where as `istio` is using [go-control-plane v0.9.6-0 ](https://github.com/istio/istio/blob/master/go.mod#L65). `endpoint_config` for EDS was [merged recently](https://github.com/envoyproxy/envoy/pull/11425) and released in v.0.9.6.

@lizan that might be the reason it was not able to provide `endpoint_config` in `config_dump` ?

File: istioctl/cmd/proxyconfig.go [hunk 1]
```diff
@@ -38,6 +38,9 @@ import (
 const (
 	jsonOutput    = "json"
 	summaryOutput = "short"
+
+	// ConigDumpPath is a path for envoy config_dump
+	ConigDumpPath = "config_dump?include_eds"
```
Comment (lizan) [inline istioctl/cmd/proxyconfig.go:43]: typo: ConfigDumpPath

**Comment by lizan** [comment]
Ah I see why, the image from istio/proxy contains the API https://github.com/envoyproxy/envoy/pull/11425, but doesn't contain the implementation https://github.com/envoyproxy/envoy/pull/11577

File: istioctl/cmd/proxyconfig.go (hunk 1 above)
Comment (shamsher31) [reply istioctl/cmd/proxyconfig.go]: Fixed, thanks.

**Comment by shamsher31** [comment]
I tired of building `istio/proxy` but faced few dependency issues (clang env) and not able to make a successful build. 

@lizan Can you help here ?

**Comment by lizan** [comment]
sure I'm working on getting 1.15.0 based istio-proxy anyway

**Comment by shamsher31** [comment]
Awesome, thanks @lizan.

File: istioctl/pkg/writer/envoy/configdump/configdump.go [hunk 2]
```diff
@@ -60,6 +60,22 @@ func (c *ConfigWriter) PrintBootstrapDump() error {
 	return nil
 }
 
+// PrintEndpointDump prints just the endpoint config dump to the ConfigWriter stdout
+func (c *ConfigWriter) PrintEndpointDump() error {
```
Comment (howardjohn) [inline istioctl/pkg/writer/envoy/configdump/configdump.go:64]: is this used anywhere?

Comment (shamsher31) [reply istioctl/pkg/writer/envoy/configdump/configdump.go:64]: I added this for testing, I will remove it.

**Comment by lizan** [comment]
tests?

**Comment by shamsher31** [comment]
@lizan @howardjohn I'm still unable to see `static_endpoint_configs` and `dynamic_endpoint_configs` in `envoyDump` during manual test.

**Comment by howardjohn** [comment]
@shamsher31 are you sure you are on latest master and using `'http://localhost:15000/config_dump?include_eds=true'`?

**Comment by shamsher31** [comment]
> @shamsher31 are you sure you are on latest master and using `'http://localhost:15000/config_dump?include_eds=true'`?

I got the response today with `static_endpoint_configs` and `dynamic_endpoint_configs`.

**Comment by shamsher31** [comment]
@howardjohn I'm thinking of refactoring `istioctl proxy-config endpoints` in the follow-up PR to use `endpoint_config` data. That requires separate config writer for endpoints so that we can extract `static_endpoint_configs` and `dynamic_endpoint_configs`.

**Comment by shamsher31** [comment]
@howardjohn @lizan @esnible

**Comment by shamsher31** [comment]
https://github.com/istio/istio/pull/27495

File: istioctl/cmd/authz.go [hunk 3]
```diff
@@ -135,8 +135,7 @@ func getConfigDumpFromPod(podName, podNamespace string) (*configdump.Wrapper, er
 	if len(pods) != 1 {
 		return nil, fmt.Errorf("expecting only 1 pod for %s.%s, found: %d", podName, podNamespace, len(pods))
 	}
-
-	data, err := kubeClient.EnvoyDo(context.TODO(), podName, podNamespace, "GET", "config_dump", nil)
+	data, err := kubeClient.EnvoyDo(context.TODO(), podName, podNamespace, "GET", ConfigDumpPath, nil)
```
Comment (howardjohn) [inline istioctl/cmd/authz.go:138]: Why does authz need eds?

**Review by howardjohn — COMMENTED** [review]
I am a bit unclear on the goal. Using EDS for the `istioctl proxy-config endpoints` makes sense, but it doesn't seem like we are doing that. It seems like we are just adding the include_eds to *all* callers (Which will increase load) but never actually doing anything with the response?

**Comment by shamsher31** [comment]
> I am a bit unclear on the goal. Using EDS for the `istioctl proxy-config endpoints` makes sense, but it doesn't seem like we are doing that.

I think I was unclear when I started working on adding `eds` to `config_dump` and added it everywhere. But I believe it is only needed for `proxy-config` and `proxy-status`. 

> It seems like we are just adding the include_eds to _all_ callers (Which will increase load) but never actually doing anything with the response?

I started working on using `EDS` for `endpoints` https://github.com/istio/istio/pull/25359#issuecomment-675891508 but lost somewhere between other work.

File: istioctl/cmd/authz.go (hunk 3 above)
Comment (shamsher31) [reply istioctl/cmd/authz.go:138]: removed.

**Comment by shamsher31** [comment]
will create another PR to use config_dump with eds.

**Pull request closed without merging**